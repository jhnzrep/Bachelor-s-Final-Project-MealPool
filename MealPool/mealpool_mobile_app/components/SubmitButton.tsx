import React from "react";
import { StyleSheet, View, Text, Image, Pressable, Alert } from "react-native";
import Colors from '../constants/Colors'
export default function SubmitButton(props: any){

  return (

    <View style={styles.buttonWrapper}>
      <Pressable
        testID="submiButton"
        onPress={props.onPress}
        style={[!props.small_button ? styles.button : styles.small_button]}
        accessibilityLabel="Learn more about this purple button">
        <Text style={[!props.small_button ? styles.button_text : styles.small_button] }>{props.text}</Text>
      </Pressable>
      <View testID="separator" style={[ props.showSeparator ? styles.alternative_login_section : {display: 'none'}]}>
          <View style={styles.separator} />
          <Text style={styles.title}>OR</Text>
          <View style={styles.separator}  />
        </View>
        <Text style={{ color: Colors.dark_orange.background }} onPress={props.navigation}>
          {props.separator_text}
      </Text>
    </View>

  );
}

const styles = StyleSheet.create({
    buttonWrapper: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      width: '100%',
      color: 'white',
      maxWidth: 300,
    },
    button: {
      backgroundColor: '#EA9B58',
      borderRadius: 25,
      width: '100%',
      height: 45,
      fontSize: 16,
      fontWeight: '600',
      color: '#FFF3DE',
      marginTop:10,
      marginBottom: 10,
      shadowColor: 'rgba(253, 180, 65, 0.3)',
      shadowOffset: { width: 5, height: 12 },
      shadowOpacity: 0.3,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    },
    small_button: {
      backgroundColor: '#EA9B58',
      borderRadius: 25,
      width: 97,
      height: 30,
      fontSize: 12,
      fontWeight: '600',
      color: '#FFF3DE',
      shadowColor: 'rgba(253, 180, 65, 0.3)',
      shadowOffset: { width: 3, height: 5 },
      shadowOpacity: 0.3,
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    },
    button_text: {
      fontWeight: '600',
      fontSize: 16,
      color: '#FFF3DE'
    },
    alternative_login_section: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      width: '100%',
      alignItems: 'center'
    },
    title: {
      fontWeight: "600",
      fontSize: 24,
      textAlign: 'center'
    },
    separator: {
      height: 1,
      width: '40%',
      backgroundColor: Colors.dark_orange.background
    },
})
