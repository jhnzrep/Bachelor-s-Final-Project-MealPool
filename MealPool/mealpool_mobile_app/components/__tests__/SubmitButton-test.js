import * as React from 'react';
import  {create, act} from 'react-test-renderer';
import { shallow } from 'enzyme';
import  renderer from 'react-test-renderer';
import { MonoText } from '../StyledText';
import SubmitButton from '../SubmitButton';
import {render, cleanup, fireEvent} from '@testing-library/react-native';

describe('<SubmitButton>', () => {
        describe('as default', () => {
        const submitbutton = render(
            <SubmitButton
                text="TestButton" 
                showSeparator={true}
                onPress={() => {}}
                navigation={() => {}} 
                separator_text="Sign in"
            />
        );

        it(`create new submit button correctly`, () => {
            const tree = renderer.create(<SubmitButton text="test button"/>).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('Renders successfully', () => {
            expect(submitbutton).toBeDefined();
        });

        it('show separator', () => {
            const rendered = render(<SubmitButton showSeparator={true}/>);
            expect(rendered.getByTestId('separator')).toBeTruthy();
        });


        it('Outputs the correct text for separator', () => {
            const rendered = render(<SubmitButton showSeparator={true} separator_text="SignIn" />);
            expect(rendered.getByText('SignIn')).toBeTruthy();
        });

        it(`submit button runs onPress`, () => {
            const onSubmit = jest.fn();
            const rendered = render(<SubmitButton onPress={ onSubmit  } text='CustomLink Component'/>);
            const buttonComponent = rendered.getByTestId('submiButton');
            fireEvent(buttonComponent, 'press');
            expect(onSubmit).toHaveBeenCalled();
        });

        it(`submit button runs onPress`, () => {
            const onSubmit = jest.fn();
            const rendered = render(<SubmitButton onPress={ onSubmit  } text='CustomLink Component'/>);
            const buttonComponent = rendered.getByTestId('submiButton');
            fireEvent(buttonComponent, 'press');
            expect(onSubmit).toHaveBeenCalled();
        });
    });
});
