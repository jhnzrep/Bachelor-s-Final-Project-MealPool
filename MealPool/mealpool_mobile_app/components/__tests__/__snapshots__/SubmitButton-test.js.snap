// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "white",
      "display": "flex",
      "justifyContent": "center",
      "maxWidth": 300,
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel="Learn more about this purple button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA9B58",
          "borderRadius": 25,
          "color": "#FFF3DE",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": "600",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "rgba(253, 180, 65, 0.3)",
          "shadowOffset": Object {
            "height": 12,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "width": "100%",
        },
      ]
    }
    testID="submiButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF3DE",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      test button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      OR
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#EA9B58",
      }
    }
  />
</View>
`;

exports[`<SubmitButton> as default create new submit button correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "white",
      "display": "flex",
      "justifyContent": "center",
      "maxWidth": 300,
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel="Learn more about this purple button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA9B58",
          "borderRadius": 25,
          "color": "#FFF3DE",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": "600",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "rgba(253, 180, 65, 0.3)",
          "shadowOffset": Object {
            "height": 12,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "width": "100%",
        },
      ]
    }
    testID="submiButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF3DE",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      test button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "none",
        },
      ]
    }
    testID="separator"
  >
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      OR
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#EA9B58",
      }
    }
  />
</View>
`;

exports[`<SubmitButton> create new submit button correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "white",
      "display": "flex",
      "justifyContent": "center",
      "maxWidth": 300,
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel="Learn more about this purple button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA9B58",
          "borderRadius": 25,
          "color": "#FFF3DE",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": "600",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "rgba(253, 180, 65, 0.3)",
          "shadowOffset": Object {
            "height": 12,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "width": "100%",
        },
      ]
    }
    testID="submiButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF3DE",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      test button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      OR
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#EA9B58",
      }
    }
  />
</View>
`;

exports[`<SubmitButton> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "white",
      "display": "flex",
      "justifyContent": "center",
      "maxWidth": 300,
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel="Learn more about this purple button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA9B58",
          "borderRadius": 25,
          "color": "#FFF3DE",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": "600",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "rgba(253, 180, 65, 0.3)",
          "shadowOffset": Object {
            "height": 12,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "width": "100%",
        },
      ]
    }
    testID="submiButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF3DE",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      test button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      OR
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#EA9B58",
      }
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "color": "white",
      "display": "flex",
      "justifyContent": "center",
      "maxWidth": 300,
      "width": "100%",
    }
  }
>
  <View
    accessibilityLabel="Learn more about this purple button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EA9B58",
          "borderRadius": 25,
          "color": "#FFF3DE",
          "display": "flex",
          "fontSize": 16,
          "fontWeight": "600",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "shadowColor": "rgba(253, 180, 65, 0.3)",
          "shadowOffset": Object {
            "height": 12,
            "width": 5,
          },
          "shadowOpacity": 0.3,
          "width": "100%",
        },
      ]
    }
    testID="submiButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF3DE",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
    >
      test button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "none",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      OR
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EA9B58",
          "height": 1,
          "width": "40%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#EA9B58",
      }
    }
  />
</View>
`;
